### PR TITLE
[Fix] Upgrade eslint and his dependencies, fix CI issues

### PR DESCRIPTION
**Pull request by ilyamore88**
Upgrades eslint dependencies and configs.

### CODE REVIEW

**Comment by robonetphy** [comment]
@ilyamore88 I remove almost all the lints except few which I didn't understand I will add the link of supported material below.

- https://github.com/microsoft/TypeScript/issues/21732
- https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md

Also in [this](https://github.com/codex-team/hawk.garage/blob/6ecbdc724f4ffe537ae83a5e7e464aa6d1ef656b/src/store/modules/events/index.ts#L1) file I have changed lot of comments so let me know if we need to change those comments.

**Review by neSpecc — APPROVED** [review]

File: src/api/user/index.js [hunk 1]
```diff
@@ -94,10 +94,10 @@ export async function updateProfile(name, email, image) {
  * @returns {Promise<boolean>}
  */
 export async function changePassword(oldPassword, newPassword) {
-  return (await api.call(MUTATION_CHANGE_PASSWORD, {
+  return api.call(MUTATION_CHANGE_PASSWORD, {
     oldPassword,
     newPassword,
-  })).changePassword;
+  }, undefined, { allowErrors: true });
```
Comment (neSpecc) [inline src/api/user/index.js:100]: in this file, I've added graphql errors support to fix the displaying them for the user. For example, 'Wrong old password entered'

File: src/components/account/settings/General.vue [hunk 2]
```diff
@@ -154,8 +154,15 @@ export default Vue.extend({
           time: 5000,
         });
       } catch (e) {
+        /**
+         * Vue18n does not support dot-notation in dict keys.
+         * As a workaround, we remove dots from message and from keys in dictionaries.
+         */
+        const dictKey = e.message.replaceAll('.', '');
```
Comment (neSpecc) [inline src/components/account/settings/General.vue:161]: fix for the displaying change-password errors. For example, 'Wrong old password entered'

File: src/components/forms/ChangePasswordFieldset.vue [hunk 3]
```diff
@@ -55,15 +55,24 @@ export default {
...
+    }
+  },
+  mounted() {
+    this.data = this.value;
+  },
   methods: {
     oldPasswordInput(value) {
-      this.value.old = value;
-      this.$emit('input', this.value);
+      this.data.old = value;
```
Comment (neSpecc) [inline src/components/forms/ChangePasswordFieldset.vue:69]: fix warning of mutation of props

File: src/i18n/index.js [hunk 4]
```diff
@@ -14,6 +14,21 @@ const defaultPluralization = VueI18n.prototype.getChoiceIndex;
...
+   * because its better to show the first choice of pluralization instead of disappearing of the whole component
+   *   — Peter, 2021 Jul 30th
+   */
+  if (!defaultPluralization){
+    const error = '( ཀ ʖ̯ ཀ) defaultPluralization failed to be defined.';
+
+    console.error(error);
+    this.vm.$sendToHawk(error)
+    return 0;
+  }
```
Comment (neSpecc) [inline src/i18n/index.js:30]: Reproduced on `/workspace/<workspace id>/settings/billing` for me

File: src/store/modules/user/index.js [hunk 5]
```diff
@@ -197,9 +197,20 @@ const actions = {
    *
```
Comment (neSpecc) [inline src/store/modules/user/index.js:197]: I've added support of change-password errors to be passed to user from the backend. For example, `Worng old password`

File: src/components/event/AssigneesList.vue [hunk 6]
```diff
@@ -49,9 +49,9 @@
 </template>
 
 <script>
-import EntityImage from '../utils/EntityImage';
-import Icon from '../utils/Icon';
-import User from '../../types/user';
+// import EntityImage from '../utils/EntityImage';
+// import Icon from '../utils/Icon';
+// import User from '../../types/user';
```
Comment (ilyamore88) [inline src/components/event/AssigneesList.vue:54]: Commented out code

File: src/components/modals/ProjectCreationDialog.vue [hunk 7]
```diff
@@ -66,7 +66,7 @@ export default {
      * @returns {Array<Workspace>} - registered workspaces with permission check
      */
     workspaces() {
-      const userId = this.$store.state.user.data.id;
+      // const userId = this.$store.state.user.data.id;
```
Comment (ilyamore88) [inline src/components/modals/ProjectCreationDialog.vue:69]: Unused code

File: src/i18n/messages/en.json [hunk 8]
```diff
@@ -131,7 +131,8 @@
       "title": "Account settings",
       "profileUpdated": "Profile was updated",
       "errors": {
-        "Wrong email format": "Wrong email format"
+        "Wrong email format": "Wrong email format 0000",
```
Comment (ilyamore88) [inline src/i18n/messages/en.json:134]:
```suggestion
        "Wrong email format": "Wrong email format",
```

**Review by robonetphy — APPROVED** [review]

File: src/utils.ts [hunk 9]
```diff
@@ -22,9 +23,11 @@ export function getEntityColor(id: string): string {
     return colors[Math.floor(Math.random() * colors.length)];
   }
 
-  const decimalId = parseInt(id.substr(-1), 16); // take last id char and convert to decimal number system
+  const decimalRadix = 16;
```
Comment (neSpecc) [inline src/utils.ts:26]: hexRadix

**Pull request merged**